classes/article/PublishedArticleDAO.inc.php

```diff
@@ -221,6 +224,9 @@ function getPublishedArticlesInSections($issueId, $useCache = false) {
 					$publishedArticles[$currSectionId]['title'] = $publishedArticle->getSectionTitle();
 				}
 			}
+			// get metadata of the current version
+			$publishedArticle = $this->getPublishedArticleByArticleId($publishedArticle->getId(), null, false, $publishedArticle->getCurrentVersionId());
```

Comment:
Hmmm... It looks like everywhere else in this class the submission revision 1 is used to get the published articles, using _fromRow without the parameters. For example in getPublishedArticlesByJournalId. Why then here only the current article version?
It seems that the concept is to get the first and then later to get the current version, or? If so maybe to do this here as well?
Maybe to take a look how those functions are used elsewhere? -- e.g. in the generic plugin WebFeedGatewayPlugin both functions getPublishedArticlesByJournalId or getPublishedArticlesInSections are used, but they would return different article versions, I suppose?
It would be even better if we could always immediately provide the latest article versions here, or?